gluten-core/src/main/scala/org/apache/spark/sql/execution/datasources/GlutenWriterColumnarRules.scala

```diff
@@ -112,7 +111,14 @@ object GlutenWriterColumnarRules {
             command.fileFormat.isInstanceOf[OrcFileFormat] =>
         if (
           BackendsApiManager.isVeloxBackend
-          && (command.partitionColumns.nonEmpty || command.bucketSpec.nonEmpty)
+          && command.partitionColumns.nonEmpty && command.staticPartitions.isEmpty
```

Comment:
we should check static partition write with `staticPartitions.size < partitionColumns.length`, not `staticPartitions.isEmpty`, as staticPartitions may not be empty while writing dynamic partitions.
```
INSERT OVERWRITE TABLE t partition(c=1, d) SELECT 3 as e
```

Reply:
@jackylee-ch Thanks for your pointing this. I have updated.

Reply:
@JkSelf is it convenient to make test for this case ?
`INSERT OVERWRITE TABLE t partition(c=1, d) SELECT 3 as e`

Reply:
@zhouyuan Added related unit tests.